Misc/osx-build-wx

```diff
@@ -83,65 +125,144 @@ while getopts "dc:t:nk:t:" opt; do
 				carbon) TOOLKIT=CARBON;;
 				cocoa)  TOOLKIT=COCOA;;
 				*)      error Unexpected toolkit: $OPTARG. Its either \"carbon\" or \"cocoa\" \(default\) 1>&2
-						;;
+				;;
 			esac
 			;;	
-        k)
-            if [ -d $OPTARG ]
-            then
-                SDKPATH=$OPTARG
-            else
-				error Invalid directory passed for SDK path: $OPTARG
-            fi
-            ;;
-        t)
-            if [ -d $OPTARG ]
-            then
-                TOOLCHAIN_DIR=$OPTARG
-            else
-				error Invalid directory passed for toolchain dir: $OPTARG
-            fi
-            ;;
+		k)
+			if [ -d $OPTARG ]
+			then
+				SDKPATH=$OPTARG
+			else
+				error "Invalid directory passed for SDK path: $OPTARG"
+			fi
+			;;
+		T)
+			if [ -d $OPTARG ]
+			then
+				TOOLCHAIN_DIR=$OPTARG
+			else
+				error "Invalid directory passed for toolchain dir: $OPTARG"
+			fi
+			;;
 		n)
 			DRY_RUN="echo "
 			;;
+        w)
+            WORKFLOW=$OPTARG
+            ;;
 		?)
-            usage $0
+			usage $0
 			exit 1
 			;;
 	esac
 done
 
+# Get the version of wxWidgets source we are building
 WX_VER=`$CONF_SCRIPT -V | grep 'wxWidgets configure' | cut -d' ' -f3`
 
-# This might be required for 10.7 & 10.8 as well
-OSX_VER=`sw_vers -productVersion`
-if [ "$OSX_VER" = "10.9" -o "$OSX_VER" \> "10.9" ]
-then
-    if [ -z "$TOOLCHAIN_DIR" ]
+#
+# This is to build the wxOSX/Cocoa version of wxWidgets, which is what is supported on macOS 10.10 and up.
+# This code tries to build wxWidgets the same way the Github workflow builds the distribution. 
+#
+doCocoaBuild() {
+
+    # If the user edited the CONF_OPTS variable, they are responsible for all of it.
+    # Othewise, we build it from the GitHub workflow file and any command line options.
+    if [ -z $CONF_OPTS ]
     then
-        # This is where gcc-4.x etc. will be found
-        TOOLCHAIN_DIR=/Developer/usr/bin
+        if [ ! -f $WORKFLOW ]
+        then
+            error "Cannot locate the workflow file: $WORKFLOW"
+        fi
+        # This is actually where 'make install' will put things.
+        WX_BUILD_DIR=$WX_PREFIX
+
+        # Fetch some parameters from the macos-latest.yaml workflow
+        if [ -z $WX_ARCH ]
+        then
+            eval `grep WX_ARCH= $WORKFLOW | sed -E -e 's/^.*"(.*)".*$/\1/'`
+        fi
+        eval `grep MACOS_VER_MIN= $WORKFLOW | sed -E -e 's/^.*"(.*)".*$/\1/'`
```

Comment:
Am I missing something or is the regex the same as on the previous line?

Reply:
The regex is the same but not the values being extracted (WX_ARCH vs. MACOS_VER_MIN).